sdks/smart-wallet-sdk/src/smartWallet.ts

```diff
@@ -0,0 +1,77 @@
+import { Interface } from '@ethersproject/abi'
+import { ChainId } from '@uniswap/sdk-core'
+
+import { abi } from '../abis/MinimalDelegation.json'
+
+import { ModeType, SMART_WALLET_ADDRESSES } from './constants'
+import { Call, MethodParameters, ExecuteOptions } from './types'
+import { ExecuteCallPlanner, ModeEncoder } from './utils'
+
+/**
+ * Main SDK class for interacting with ERC7821-compatible smart wallets
+ */
+export class SmartWallet {
+  /**
+   * Interface for the Smart Wallet contract (will be initialized from ABI)
+   */
+  public static INTERFACE: Interface = new Interface(abi)
+
+  /**
+   * Creates method parameters for executing a batch of calls through a smart wallet
+   * @param calls Array of calls to encode
+   * @param options Execution options
+   * @param chainId The chain ID for the calls
+   * @returns Method parameters with calldata and value
+   */
+  public static encodeExecute(calls: Call[], options: ExecuteOptions = {}): MethodParameters {
+    const mode = this.getModeFromOptions(options)
+    const planner = new ExecuteCallPlanner()
+    for (const call of calls) {
+      planner.add(call.to, call.data, call.value)
+    }
+    // Add dummy opData for modes that require it
```

Comment:
I'm confused on how this is used. Why do you need to loop through all the calls and add them to the planner. Can't you just initialize the planner with the array?